src/items.zig

```diff
@@ -157,47 +157,48 @@ const MaterialProperty = enum {
 	}
 };
 
-pub const BaseItemIndex = packed struct {
-	index: u16,
+pub const BaseItemIndex = enum(u16) {
+	no_value = std.math.maxInt(u16),
```

Comment:
Should be added when needed.
Especially since adding this would require checking (and knowing that you have to check) it in every use-cases.

> As for optional vs no_value, optional in this case means 25 extra bytes on every tool, no_value reduces ID range, but has a very low memory cost.

The big big advantage of optional, is that it doesn't hide the fact that it is in fact optional.
For in-memory-storage these extra bytes do not matter at all (and it would be 50 bytes even, due to alignment).
And for in-file storage or networking there are better ways to compress the data anyways.